app/hooks/useQueryParam.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  useState,
+  useEffect,
+  useRef,
+  useLayoutEffect,
+  useCallback,
+} from "react";
```

Comment:
Going for the full set ;)

@jishaal - do you prefer `React.use...` or the separate imports here?

Reply:
Hehe, I prefer this way personally though no strong opinion on either. Just less words haha